android/src/main/java/com/reactnativebledidcomm/BleDidcommModule.kt

```diff
@@ -10,30 +10,59 @@ import androidx.annotation.RequiresPermission
 import com.facebook.react.bridge.*
 import com.facebook.react.modules.core.DeviceEventManagerModule
 import com.reactnativebledidcomm.central.CentralManager
+import com.reactnativebledidcomm.central.CentralManagerException
 import com.reactnativebledidcomm.peripheral.PeripheralManager
+import com.reactnativebledidcomm.peripheral.PeripheralManagerException
 import java.util.*
 
 
 class BleDidcommModule(private val context: ReactApplicationContext) :
     ReactContextBaseJavaModule(context) {
-    private lateinit var centralManager: CentralManager
-    private lateinit var peripheralManager: PeripheralManager
+    private var centralManager: CentralManager? = null
+    private var peripheralManager: PeripheralManager? = null
 
     override fun getName(): String {
         return Constants.TAG
     }
 
     @ReactMethod
-    @Suppress("unused")
     fun startCentral(
+        @Suppress("UNUSED_PARAMETER") options: ReadableMap,
+        promise: Promise
+    ) {
+        try {
+            this.centralManager = CentralManager(context)
+            promise.resolve(null)
+        } catch (e: Exception) {
+            promise.reject("error", e)
+        }
+    }
+
+    @ReactMethod
+    @RequiresPermission(value = "android.permission.BLUETOOTH_CONNECT")
```

Comment:
is BLUETOOTH_CONNECT the only permissions required?

Reply:
For this method, yes.